sql/src/main/java/io/crate/operation/projectors/fetch/FetchProjectorContext.java

```diff
@@ -67,7 +67,7 @@ ReaderBucket readerBucket(int readerId) {
     }
 
     ReaderBucket require(long doc) {
-        int readerId = (int) (doc >> 32);
+        int readerId = (int) ((doc >> 32) & 0x00ffffffL);
```

Comment:
should we have a FetchId class with static methods? E.g. `extractReaderId(long fetchId)`